hack/update-codegen.sh

```diff
@@ -24,28 +24,21 @@ CODEGEN_PKG=$($GO_CMD list -m -f "{{.Dir}}" k8s.io/code-generator)
 
 cd $(dirname ${BASH_SOURCE[0]})/..
 
-chmod +x "${CODEGEN_PKG}/generate-internal-groups.sh"
-chmod +x "${CODEGEN_PKG}/generate-groups.sh"
-"${CODEGEN_PKG}/generate-groups.sh" \
-  applyconfiguration,client,lister,informer \
-  sigs.k8s.io/kueue/client-go \
-  sigs.k8s.io/kueue/apis \
-  kueue:v1beta1 \
-  --go-header-file ${KUEUE_ROOT}/hack/boilerplate.go.txt
-
-
-# Future releases of of code-generator add better support for out of GOPATH generation,
-# check https://github.com/kubernetes/code-generator/blob/master/examples/hack/update-codegen.sh for details,
-# for now we should just move the generated code to `client-go`.
-if  [ -d sigs.k8s.io/kueue/client-go ]
-then
-  echo "Out of GOPATH generation, move the generated files to ../client-go"
-  if  [ -d ${KUEUE_ROOT}/client-go ]
-  then
-    rm -r ${KUEUE_ROOT}/client-go
-  fi
-  mv sigs.k8s.io/kueue/client-go ${KUEUE_ROOT}/client-go
-fi
+chmod +x "${CODEGEN_PKG}/kube_codegen.sh"
+source "${CODEGEN_PKG}/kube_codegen.sh"
+
+kube::codegen::gen_helpers \
+  --input-pkg-root sigs.k8s.io/kueue/apis \
+  --output-base "${KUEUE_ROOT}/../../" \
+  --boilerplate ${KUEUE_ROOT}/hack/boilerplate.go.txt
+
+kube::codegen::gen_client \
+  --input-pkg-root sigs.k8s.io/kueue/apis \
+  --output-pkg-root sigs.k8s.io/kueue/client-go \
+  --output-base "${KUEUE_ROOT}/../../" \
```

Comment:
I think this assumes that the code is loacted in `<some-path>/sigs.k8s.io/kueue`, can you tray 

```suggestion
  --output-pkg-root client-go \
  --output-base "${KUEUE_ROOT}" \
```

Reply:
Indeed this is the assumption, however changing that results in change of the name of the generated package

Reply:
Let me investigate it further

Reply:
According to this [documentation](https://github.com/kubernetes/code-generator/blob/791c213a776b4b92f9af0805c1cf57c118ba951c/kube_codegen.sh#L389) ```--output-pug-root``` must be Go package syntax which in our case is ```sigs.k8s.io/kueue/client-go```. What I can add to prevent misusing this script is a check that generates a short warning/error if ```kueue/``` is not a subdirectory of ```sigs.k8s.io/```. WDYT @trasc

Reply:
That will be problematic, at least for me personally, I'll not be able to do anything on the project unless I change the way I structure my work dirs for years. :) 

Are we sure that there is no way around this?

Reply:
@trasc IIUC the previous code also assumed that kueue is inside `sigs.k8s.io`, didn't it? If this is the case it wouldn't be a regression.

Reply:
It will be a regression on our side, the previous script used to work outside of GOPATH but would generate the files in a different dir, hence the moving hack.

Reply:
cloud-provider-gsp looks to be trying to accomplish this, however I did not looked into details on what they are doing.
https://github.com/kubernetes/cloud-provider-gcp/blob/32ba3c7b4240d23c0cfc55d57d5dd50bccbb87c7/crd/hack/generate-groups.sh#L52

Reply:
I played with this today a little. 

First, I reproduced the issue with the client-go generation when the kueue is in a different folder than `sigs.k8s.io`. 

Second, I was able to resolve the issue with this:
```sh
PARENT_DIR=$(basename $(dirname ${KUEUE_ROOT}))
...
kube::codegen::gen_helpers \
  --input-pkg-root "${PARENT_DIR}/kueue/apis" \
...
kube::codegen::gen_client \
  --input-pkg-root "${PARENT_DIR}/kueue/apis" \
  --output-pkg-root "${PARENT_DIR}/kueue/client-go" \
...
```
@trasc, do you think this addresses the issue? @PBundyra please update the PR accordingly if you double-check and this works.

Reply:
Ok, I think I found a solution that works even if kueue is out of sigs.k8s.io, but using a soft link to force it :)
```sh
ln -s .. sigs.k8s.io

kube::codegen::gen_helpers \
  --input-pkg-root sigs.k8s.io/kueue/apis \
  --output-base "${KUEUE_ROOT}" \
  --boilerplate ${KUEUE_ROOT}/hack/boilerplate.go.txt

kube::codegen::gen_client \
  --input-pkg-root sigs.k8s.io/kueue/apis \
  --output-pkg-root sigs.k8s.io/kueue/client-go \
  --output-base "${KUEUE_ROOT}" \
  --boilerplate ${KUEUE_ROOT}/hack/boilerplate.go.txt \
  --with-watch \
  --with-applyconfig

rm sigs.k8s.io
```
This can be refined to make the trick with soft link conditional only if we are out of `sigs.k8s.io`, but not sure it is worth it.

Reply:
Or even better we can have something like here: https://github.com/kubernetes-sigs/kueue/commit/70b3bd3077bdabb23ad4346ce415f8bc35cec6ef. Feel free to cherry-pick the commit on your branch.

Reply:
LGTM